pkg/cmd/search/repos/repos.go

```diff
@@ -0,0 +1,198 @@
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.GoodFirstIssues, "good-first-issues", "", "Filter on number of issues with the 'good first issue' label")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.HelpWantedIssues, "help-wanted-issues", "", "Filter on number of issues with the 'help wanted' label")
+	cmdutil.StringSliceEnumFlag(cmd, &opts.Query.Qualifiers.In, "in", "", nil, []string{"name", "description", "readme"}, "Restrict search to specific part of repository")
+	cmd.Flags().StringSliceVar(&opts.Query.Qualifiers.Language, "language", nil, "Filter based on the coding language")
+	cmd.Flags().StringSliceVar(&opts.Query.Qualifiers.License, "license", nil, "Filter based on license type")
+	cmdutil.NilBoolFlag(cmd, &opts.Query.Qualifiers.Mirror, "mirror", "", "Filter based on mirror state")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Org, "org", "", "Filter on organization")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Pushed, "updated", "", "Filter on last updated at date")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Repo, "repo", "", "Filter on repository name")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Size, "size", "", "Filter on a size range, in kilobytes")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Stars, "stars", "", "Filter on number of stars")
+	cmd.Flags().StringSliceVar(&opts.Query.Qualifiers.Topic, "topic", nil, "Filter on topic")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Topics, "number-topics", "", "Filter on number of topics")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.User, "user", "", "Filter based on user")
+	cmdutil.StringEnumFlag(cmd, &opts.Query.Qualifiers.Is, "visibility", "", "", []string{"public", "private"}, "Filter based on visibility")
+
+	return cmd
+}
+
+func reposRun(opts *ReposOptions) error {
+	io := opts.IO
+	if opts.Searcher == nil {
+		cfg, err := opts.Config()
+		if err != nil {
+			return err
+		}
+		host, err := cfg.DefaultHost()
+		if err != nil {
+			return err
+		}
+		client, err := opts.HttpClient()
+		if err != nil {
+			return err
+		}
+		opts.Searcher = search.NewSearcher(client, host)
+	}
+	if opts.WebMode {
+		url := opts.Searcher.URL(opts.Query)
+		if io.IsStdoutTTY() {
+			fmt.Fprintf(io.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(url))
+		}
+		return opts.Browser.Browse(url)
+	}
+	io.StartProgressIndicator()
+	result, err := opts.Searcher.Repositories(opts.Query)
+	io.StopProgressIndicator()
+	if err != nil {
+		return err
+	}
+	if err := io.StartPager(); err == nil {
+		defer io.StopPager()
+	} else {
+		fmt.Fprintf(io.ErrOut, "failed to start pager: %v\n", err)
+	}
+	if opts.Exporter != nil {
+		return opts.Exporter.Write(io, result.Items)
+	}
+	return displayResults(io, result)
+}
+
+func displayResults(io *iostreams.IOStreams, results search.RepositoriesResult) error {
+	cs := io.ColorScheme()
+	tp := utils.NewTablePrinter(io)
+	for _, repo := range results.Items {
+		var tags []string
+		if repo.Private {
```

Comment:
Note that "internal" repositories also show up as `private: true`, but their visibility should be consulted and displayed as "internal", i.e. distinct from "private". https://github.com/cli/cli/issues/4151